### PR TITLE
fix Layer.__dir__  to show the parameters/sublayers/buffers/attr/method of the Layer

### PR DESCRIPTION
**Pull request by zhwesky2010**
### PR types

New features
### PR changes

APIs
### Describe

New API: 
```
Layer.__dir__
```
modify dir() to make it easy for users to view the attr/parameters/sublayers/function of the Layer.

For example:

```
import paddle.fluid as fluid
import numpy as np

fluid.dygraph.enable_dygraph()

class Mylayer(fluid.dygraph.Layer):
    def __init__(self):
        super(Mylayer, self).__init__()
        self.linear1 = fluid.dygraph.Linear(10, 10)
        self.linear2 = fluid.dygraph.Linear(5, 5)
        self.conv2d = fluid.dygraph.Conv2D(3, 2, 3)
        self.embedding = fluid.dygraph.Embedding(size=[128, 16])
        self.h_0 = fluid.dygraph.to_variable(np.zeros([10, 10]).astype('float32'))

mylayer = Mylayer()
print(dir(mylayer))
```

image

### CODE REVIEW

File: python/paddle/fluid/dygraph/layers.py [hunk 1]
```diff
@@ -752,6 +752,40 @@ def __delattr__(self, name):
...
+                def __init__(self):
+                    super(Mylayer1, self).__init__()
+                    self.linear1 = fluid.dygraph.Linear(10, 10)
+                    self.linear2 = fluid.dygraph.Linear(5, 5)
+                    self.conv2d = fluid.dygraph.Conv2D(3, 2, 3)
+                    self.embedding = fluid.dygraph.Embedding(size=[128, 16])
+                    self.h_0 = fluid.dygraph.to_variable(np.zeros([10, 10]).astype('float32'))
+
+            mylayer = Mylayer()
+            print(dir(mylayer))
```
Comment (zhiqiu) [inline python/paddle/fluid/dygraph/layers.py:775]: Better add the output of the example code

Comment (zhwesky2010) [reply python/paddle/fluid/dygraph/layers.py:775]: Ok

Comment (zhwesky2010) [reply python/paddle/fluid/dygraph/layers.py:775]: Done

**Review by zhiqiu — APPROVED** [review]
LGTM

**Pull request merged**